desafio-07/flafmg/csharp/InvertFile.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.IO;
+using System.Text;
+using System.Collections.Generic;
+
+class Program
+{
+    const int BlockSize = 100;
+    const int BufferSize = BlockSize * 1024 * 1024;
+
+    static void Main(string[] args)
+    {
+        if (args.Length != 1)
+        {
+            Console.WriteLine("argumentos vazios, passe o local do arquivo nos argumentos");
+            return;
+        }
+
+        string filePath = args[0];
```

Comment:
Prefira usar inferência de tipo (`var`)

Reply:
Acho uso excessivo de var confuso, prefiro evitar (mania de javeiro eu acho)

Reply:
Inferência de tipo foi introduzida ao C# em 2007, hoje não é apenas uma feature mas uma "regra não escrita", praticamente todas as codebases usam massivamente inferência de tipo (enquanto no Java foi introduzido apenas em 2018, logo ainda há codebases que não o utilizam). Enquanto inferência de tipo não deixa claro qual o tipo você está recebendo ela te estimula a dar nomes melhores às suas variáveis, expressando o que elas são ou significam. Muitas vezes isso já vai ser suficiente para, mais tarde no código, você inferir o que aquilo é. `clientsList`, `creditCardType` ou `maximumInstallments` deixam claro, sem precisar saber o tipo, a que se referem. No mais, tome por exemplo:
```
...
List<Client> clientList = someService.GetClients();
CreditCardType type = clientList.FirstOrDefault(s => s.Id == clientId).CreditCard.Type;
int maximumInstallments = shoppingCart.GetMaximumInstallments();
```
Versus:
```
...
var clientList = someService.GetClients();
var type = clientList.FirstOrDefault(s => s.Id == clientId).CreditCard.Type;
var maximumInstallments = shoppingCart.GetMaximumInstallments();
```
Se em um você precisa escanear as colunas de código para procurar o nome de cada variável e entender seu contexto, no outro todos os nomes das variáveis começam na mesma coluna.
Como disse, quando for trabalhar com C# você encontrará em muito mais abundância o uso de inferência de tipo (como no segundo exemplo) que tipagem explícita, exceto em alguns casos bastante específicos como comunicação com APIs C/C++ (onde você precisa especificar se irá mandar/receber um `Int32` ou `Int16`).

Reply:
Mas eu não deveria programar da forma que me seja mais confortável? Contanto que não faça meu código ilegível e com péssima performance? Usar explícito ou inferência no final das contas nao altera a performance, talvez fique levemente mais difícil de ler porque os nomes das variáveis nao vão estar alinhados mas não creio que seja o fim do mundo
Irei tentar aderir a sugestão mas eu realmente não sou muito fã de tipo iferido, até em Rust eu tento usar o tipo iferido mas sempre me pego botando explícito um mut let x: Int 
Talvez seja mais uma questão de tentar me forçar a usar até pegar costume

Reply:
>Mas eu não deveria programar da forma que me seja mais confortável?
Sim e não. O computador leva milésimos de segundo para ler o código que escrevemos mas se escrevemos algo difícil de entender ou que fuja dos padrões ficará difícil para outra pessoa entender e essa é a parte problemática - se estiver difícil para outras pessoas entenderem, será mais difícil evoluir e dar manutenção no código. Existem padrões para cada linguagem, se você escreve dentro dos padrões outras pessoas que escrevem código naquela linguagem ao lerem seu código terão facilidade em entende-lo, mas se você escreve fora dos padrões, trazendo costumes e estilos de outras linguagens, ficará mais difícil. Não é uma questão de X ser melhor que Y e sim de seguir os padrões de cada linguagem - eu tenho certeza que boa parte dos programadores C não gostam muito quando veem código C# com 120 colunas, mas isso é porque a comunidade C lida com indentação e nomenclaturas de um modo diferente da comunidade C#; ou como em Java ao se declarar uma "variável constante" se utiliza UPPER_SNAKE_CASE enquanto no C# se utiliza PascalCase - alguém lendo um código muito para dentro de uma classe que utiliza uma variável assim nomeada facilmente identificaria, em cada linguagem, se aquela é uma variável constante ou não mas se você utiliza um estilo de outra linguagem isso poderia gerar confusão.

Reply:
ok, consegui pegar um computador emprestado rapidinho ja que o meu infelizmente quebrou o monitor e to sem tempo muito menos dinheiro pra ajeitar kskk, eu vou fazer as atualizações no codigo e envio jaja, vou aproveitar e atualizar de todos os outros codigos pra ja evitar a fadiga